core/extractors/extractors_latex.py

```
@@ -0,0 +1,528 @@
+            return self._dict_to_result(self._empty_result(latex_path, str(e)))
+    
+    def _is_safe_tar_member(self, member: tarfile.TarInfo, target_path: Path) -> bool:
+        """Check if a tar member is safe to extract (no path traversal)."""
+        # Reject absolute paths
+        if member.name.startswith('/') or member.name.startswith('\\'):
+            return False
+        # Reject paths with .. that could escape target
+        member_path = (target_path / member.name).resolve()
+        try:
+            member_path.relative_to(target_path.resolve())
+            return True
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)